### DISCUSSION TITLE
Database Real Time CDC table joins

### DISCUSSION THREAD

**Discussion by briosheje** [supabase/supabase#9501] [Questions] [answered]
I have a scenario where I'm looking for realtime events on a table and updates my application state accordingly.

However, in one specific scenario, I do a join when I query the data, like this:

```
const { data } = await supabase.from('table').select('*, joined_table');
```

When I subscribe to realtime changes on `table`, in `payload.new` and `payload.old` I get back the added/updated/removed records, but it would be awesome to be able to specify in the `from` statement what needs to be joined.

In my current case, when I get a realtime change event, I fetch the record using `payload.new.[primarykey]` with the join syntax as specified above and then update the state, but I was wondering whether there is a way to specify what to join like you can specify what to filter like in the following example:

``` 
// Supabase client setup

const channel = supabase
  .channel('db-changes')
  .on(
    'postgres_changes',
    { event: '*', schema: 'public', table: 'messages', filter: 'body=eq.bye' },
    (payload) => console.log(payload)
  )
  .on(
    'postgres_changes',
    { event: 'INSERT', schema: 'public', table: 'users' },
    (payload) => console.log(payload)
  )
  .subscribe()
```

^-- taken from https://supabase.com/docs/guides/realtime/postgres-cdc

**Comment by GaryAustin1** [comment, marked as answer]
There is no support for joins or views in realtime.  You either have to do the extra select with a join, or fetch the join data and merge it yourself on the change event.  Depending on what you are joining you can also add that data to your realtime monitored table rows, but that is probably a rare case to be reasonable.

**Reply by briosheje** [reply]
Okay perfect, that makes sense Indeed. Thanks
